blade-graphics/src/vulkan/resource.rs

```diff
@@ -83,6 +156,35 @@ impl super::Context {
         }
     }
 
+    pub fn get_texture_fd(&self, texture: super::Texture) -> isize {
+        let mut manager = self.memory.lock().unwrap();
+        let block = manager
+            .slab
+            .get(texture.memory_handle)
+            .expect("get_texture_fd: Invalid memory_handle");
+
+        //TODO make non-windows variant
+        #[cfg(target_os = "windows")]
+        {
+            let get_handle_info = vk::MemoryGetWin32HandleInfoKHR {
+                handle_type: vk::ExternalMemoryHandleTypeFlags::OPAQUE_WIN32_KMT,
```

Comment:
any reason this is hardcoded to KMT?

Reply:
Ah, i added this quickly so i could test in the mirror example, but i should indeed provide a way for each external type.

Reply:
If its alright for the return value I'm gonna use `ExternalMemorySource` although its a bit off that it returns it with always `Some` in the enums. It would at least make it so there doesn't have to be another enum full off platform specific configurations.